hack/images/kicbase.Dockerfile

```diff
@@ -34,12 +34,21 @@ USER root
 # https://github.com/kubernetes-sigs/kind/blob/master/images/base/files/usr/local/bin/entrypoint
 RUN mkdir -p /kind
 RUN rm -rf \
-    /var/cache/debconf/* \
-    /var/lib/apt/lists/* \
-    /var/log/* \
-    /tmp/* \
-    /var/tmp/* \
-    /usr/share/doc/* \
-    /usr/share/man/* \
-    /usr/share/local/* \
-RUN echo "kic! Build: ${COMMIT_SHA} Time :$(date)" > "/kic.txt"
+  /var/cache/debconf/* \
+  /var/lib/apt/lists/* \
+  /var/log/* \
+  /tmp/* \
+  /var/tmp/* \
+  /usr/share/doc/* \
+  /usr/share/man/* \
+  /usr/share/local/* \
+  RUN echo "kic! Build: ${COMMIT_SHA} Time :$(date)" > "/kic.txt"
+
+
+FROM busybox
+ARG KUBERNETES_VERSION
+COPY out/preloaded-images-k8s-$KUBERNETES_VERSION.tar /preloaded-images.tar
+RUN tar xvf /preloaded-images.tar -C /
+
+FROM base
+COPY --from=1 /var/lib/docker /var/lib/docker 
```

Comment:
This is a clever way to avoid coding specific versions of images to docker pull. Does `docker ps` actually show these images?

Also, we should copy `/var/lib/minikube/binaries`. Feel free to save that for a follow-up PR if you prefer.

Reply:
What happens to containerd images ?

Reply:
I wonder if there is a way we give the images to crictl and let both docker and containerd use it?

Reply:
maybe using crctl we could make it available for all runtimes ?

Reply:
I was going to handle crictl and containerd in a separate PR, they may or may not have the same storage structure for images.

Does `/var/lib/minikube/binaries` exist in the current kic base image? If so then it should be there in the preloaded image as well.

Reply:
Currently, `docker ps` shows:

`gcr.io/k8s-minikube/kicbase:v0.0.5-k8s-v1.17.2` if using the preloaded image or

`gcr.io/k8s-minikube/kicbase:v0.0.5` (current default) if using a k8s version that doesn't have a preloaded image

Reply:
thanks for converting it to a go script, 

heads up "/var/lib/minikube/binaries" is going to be changed to "/usr/local/minikube/..."
it is better to use the vmpath.GuestPersistentDir

or also if we run minikube start it should create all those stuff for us in it so we dont need to depend on that constant

Reply:
I am okay with shipping pre-load first for our docker run time then think about others later. as most ppl just go with the default one